enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,964 @@
+  // and defined as the failure domain.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  Datacenter string `json:"datacenter"`
+
+  // computeCluster as the failure domain
+  // This is required to be a path
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  ComputeCluster *string `json:"computeCluster,omitempty"`
+
+  // Hosts has information required for placement of machines on VSphere hosts.
+  // +optional
+  Hosts *VSpherePlatformFailureDomainHosts `json:"hosts,omitempty"`
+
+  // networks is the list of networks within this failure domain
+  // +kubebuilder:validation:Optional
+  Networks []string `json:"networks,omitempty"`
+
+  // datastore is the name or inventory path of the datastore in which the
+  // virtual machine is created/located.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  Datastore string `json:"datastore,omitempty"`
+}
+
+// FailureDomainHosts
+type FailureDomainHosts struct {
+  // vmGroupName is the Virtual Machine Group name configured
+  // within a vCenter cluster that is associated with
+  // the corresponding Host Group.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  VMGroupName string `json:"vmGroupName"`
+
+  // hostGroupName is the Host Gorup name configured
+  // within a vCenter cluster defining a group
+  // of ESXi hosts.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields
+// https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go
+type VCenterSpec struct {
+
+  // server is the fully-qualified domain name or the IP address of the vCenter server.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=255
+  Server string `json:"server"`
+
+  // port is the TCP port that will be used to communicate to
+  // the vCenter endpoint. This is typically unchanged from
+  // the default of HTTPS TCP/443.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=32767
+  // +kubebuilder:default=443
+  Port uint `json:"port,omitempty"`
+
+  // User is the username that will be used to connect to vCenter
+  // +kubebuilder:validation:Required
+  User string `json:"user"`
+
+  // Password is the password for the user to use to connect to the vCenter.
+  // +kubebuilder:validation:Required
+  Password string `json:"password"`
+
+  // Datacenter in which VMs are located.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinItems=1
+  Datacenters []string `json:"datacenters"`
```

Comment:
What happens if a datacenter is specified here but not specified in failure-domain topology? Or what happens if a datacenter that exists in failure-domain topology does not exist here? Are both the fields meant to have same aggregated values?

Reply:
@rvanderp3 has us covered



[https://github.com/openshift/installer/pull/5911/commits/b897299813e258c3cb4a7eb0d7460c12e64451b5#diff-2cdb5d09122e1d2650[…]c5f41c9ae005febR268-R280](https://github.com/openshift/installer/pull/5911/commits/b897299813e258c3cb4a7eb0d7460c12e64451b5#diff-2cdb5d09122e1d26502732ec3ee5fb818c7153af73b55c95cc5f41c9ae005febR268-R280)
https://github.com/openshift/installer/blob/b897299813e258c3cb4a7eb0d7460c12e64451b5/pkg/types/vsphere/validation/platform_test.go#L398-L406